docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,141 @@
+  (usually configuration files, but not necessarily),
+  that will be managed from the cloud tenant.
+* Notably, __the plugin configuration itself is managed from the cloud__,
+  meaning, the device owner can update from the cloud the list of files to be managed.
+* By default, __the managed files are represented on the cloud by their full path on the device__.
+  Optionally, the device owner can assign an alias to each target path.  
+* When files are downloaded from the cloud to the device,
+  __these files are stored in a temporary directory first__.
+  They are atomically moved to their target path, only after a fully successful download.
+  The aim is to avoid breaking the system with half downloaded files.
+* Once all the files of a snapshot have been downloaded,
+  __the plugin published a notification message on the local thin-edge MQTT bus__.
+  The device software has to subscribe to these messages if any action is required,
+  say to check the content of file, to preprocess it or to restart a daemon. 
+* In other words, the responsibilities of the plugin are:
+  * to define the list of files under configuration management
+  * to notify the cloud when this list is updated,
+  * to upload these files to the cloud on demand,  
+  * to download the files pushed from the cloud,
+  * to make sure that the target files are updated atomically after successful download,
+  * to notify the device software when the configuration is updated.
+* By contrast, the plugin is not responsible for:
+  * checking the uploaded files are well-formed,
+  * restarting the configured processes.
+* A user-specific component, installed on the device,
+  can implement more sophisticated configuration use-cases by:
+  * listening for configuration updates on the local thin-edge MQTT bus,
+  * restarting the appropriate processes when appropriate,  
+  * declaring intermediate files as the managed files,
+    to have the opportunity to check or update their content
+    before moving them to the actual targets.
+
+## Installation
+
+Assuming the configuration plugin `c8y_configuration_plugin`
+has been installed in `/usr/bin/c8y_configuration_plugin`,
+two files must be added under `/etc/tedge/operations/c8y/`
+to declare that this plugin supports two Cumulocity operations:
+uploading and downloading configuration files
+(which respective SmartRest2 codes are `526` and `524`).
+
+```shell
+$ ls -l /etc/tedge/operations/c8y/c8y_UploadConfigFile
+-rw-r--r-- 1 tedge tedge 95 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_UploadConfigFile
+  
+$ ls -l /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+-rw-r--r-- 1 tedge tedge 97 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+```
+
+The `c8y_configuration_plugin` has to be run as a daemon on the device, the latter being connected to Cumulocity.
+
+## Configuration
+
+The `c8y_configuration_plugin` configuration is stored by default under `/etc/tedge/c8y/c8y_configuration_plugin.toml`
```

Comment:
I needed to change the file name to `c8y-configuration-plugin.toml` (with `-`) because c8y didn't show any files under "AVAILABLE SUPPORTED CONFIGURATIONS" with `_`. I think it's a bug of c8y, however, the easiest solution was to change the file name from underscore to hyphen.